### ISSUE TITLE
MASS GUI components need to build *.ui to *_ui.py

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#91]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/91) by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

This has been a bug for years. The installer should do the equivalent of

``` 
pyuic4-2.7 load_data_dialog_form.ui > load_data_dialog_form_ui.py
pyuic4-2.7 make_cuts_dialog_form.ui > make_cuts_dialog_form_ui.py
```

during the build phase (i.e., before installing). I have never figured out how to make distutils do this, but I really need to. @kelseymorgan found that this was blocking her work with Daikang this week.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

See [pyqt-distutils](https://pypi.python.org/pypi/pyqt-distutils/) for the package that perhaps solves this. ??

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Or [distutils_ui](https://pypi.python.org/pypi/distutils_ui) might be more modern.

BUT perhaps it's better just to have normal distutils call a shell command or a make command, or whatever, so that we don't need end-users to install pyqt4-distutils or distutils-ui. Probably that's better.

OR maybe there's simply a bug in the `except` clause of the make_cuts_dialog.py file:

```
# Import the form created by QtDesigner.  If possible, use a *.py file created
# with the pyuic command-line tool and import as usual....
try:
    from make_cuts_dialog_form_ui import Ui_Dialog

# ...though this is not always possible.  If necessary, use the PyQt4.uic package.
# In fact, I have not figured out how to have distutils convert the *.ui to a *.py
# file automatically, so the above will basically always fail.  Hmmm.

except ImportError:
    import PyQt4.uic
    path,_ = os.path.split(__file__)
    ui_filename = os.path.join(path, "make_cuts_dialog_form.ui")
    Ui_MakeCuts, _load_data_dialog_baseclass = PyQt4.uic.loadUiType(ui_filename)
```

Does the last Ui_MakeCuts need to be Ui_Dialog ?

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Fix bug/typo in mass.gui.create_cuts().

Fixes #91.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Compile Qt user interface files into their python wrappers.

Further fixes #91. Note that this un-does commit 0e8e3eb10 from 18
months ago, so it’s possible that this will cause problems.
